cmd/broker/ingress/main.go

```diff
@@ -0,0 +1,158 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/pkg/provisioners"
+	"github.com/knative/pkg/signals"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	port = 8080
+
+	readTimeout  = 1 * time.Minute
+	writeTimeout = 1 * time.Minute
+)
+
+func main() {
+	logConfig := provisioners.NewLoggingConfig()
+	logger := provisioners.NewProvisionerLoggerFromConfig(logConfig).Desugar()
+	defer logger.Sync()
+	flag.Parse()
+
+	logger.Info("Starting...")
+
+	mgr, err := manager.New(config.GetConfigOrDie(), manager.Options{})
+	if err != nil {
+		logger.Fatal("Error starting up.", zap.Error(err))
+	}
+
+	if err = eventingv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		logger.Fatal("Unable to add eventingv1alpha1 scheme", zap.Error(err))
+	}
+
+	c := getRequiredEnv("CHANNEL")
+
+	h := NewHandler(logger, c)
+
+	s := &http.Server{
+		Addr:         fmt.Sprintf(":%d", port),
+		Handler:      h,
+		ErrorLog:     zap.NewStdLog(logger),
+		ReadTimeout:  readTimeout,
+		WriteTimeout: writeTimeout,
+	}
+
+	err = mgr.Add(&runnableServer{
+		logger: logger,
+		s:      s,
+	})
+	if err != nil {
+		logger.Fatal("Unable to add runnableServer", zap.Error(err))
+	}
+
+	// Set up signals so we handle the first shutdown signal gracefully.
+	stopCh := signals.SetupSignalHandler()
+	// Start blocks forever.
+	if err = mgr.Start(stopCh); err != nil {
+		logger.Error("manager.Start() returned an error", zap.Error(err))
+	}
+	logger.Info("Exiting...")
+
+	ctx, cancel := context.WithTimeout(context.Background(), writeTimeout)
+	defer cancel()
+	if err = s.Shutdown(ctx); err != nil {
+		logger.Error("Shutdown returned an error", zap.Error(err))
+	}
+}
+
+func getRequiredEnv(envKey string) string {
+	val, defined := os.LookupEnv(envKey)
+	if !defined {
+		log.Fatalf("required environment variable not defined '%s'", envKey)
+	}
+	return val
+}
+
+// http.Handler that takes a single request in and sends it out to a single destination.
+type Handler struct {
+	receiver    *provisioners.MessageReceiver
+	dispatcher  *provisioners.MessageDispatcher
+	destination string
+
+	logger *zap.Logger
+}
+
+// NewHandler creates a new ingress.Handler.
+func NewHandler(logger *zap.Logger, destination string) *Handler {
```

Comment:
newHandler? Or move it to separate package and keep it as it is?

Reply:
I intended to move this out to somewhere in `pkg/`. That will be done as #897.